mautic/mautic
# WIP : Better UI for relative date segment filters

| Q                                      | A
| -------------------------------------- | ---
| Bug fix? (use the a.b branch)          | No
| New feature/enhancement? (use the a.x branch)      | Yes
| Deprecations?                          | No
| BC breaks? (use the c.x branch)        | No
| Automated tests included?              | Yes 
| Related user documentation PR URL      | mautic/mautic-documentation#... 
| Related developer documentation PR URL | mautic/developer-documentation#... 
| Issue(s) addressed                     | Fixes #... 



#### Description:
We have added "absolute" and "relative" date values for date and datetime segment filter.
**Note:** Added a option to write relative date to the absolute date input to keep backward compatibility.

Remove Unnecessary Absolute and relative filter from Segments.
Change filter field type to textbox for input string for like, not like, regex, not regex, starts with, ends with and contains.
We should have validation where customer should only able to add positive numeric value for relative type date filter.

Absolute Date:
![image](https://github.com/mautic/mautic/assets/68939488/8505eba8-35c3-4d11-a44c-337d71b20883)

Relative Date:
![image](https://github.com/mautic/mautic/assets/68939488/63a1f071-3c4a-450f-8635-3723f24c0602)


#### Steps to test this PR:


1. Open this PR on Gitpod or pull down for testing locally (see docs on testing PRs [here](https://contribute.mautic.org/contributing-to-mautic/tester))
2. Open this PR on Gitpod or pull down for testing locally (see docs on testing PRs [here](https://contribute.mautic.org/contributing-to-mautic/tester))
3. When date or datetime filed is select for segment filter it should show the absolute and relative date fields.
4. Old date filters in segment should work while segment building.
5. Verify that customer should only able to add positive numeric value for relative type date filter.
6. Also test Absolute and Relative date filter with greater than, greater than equals, less than and less than equals operators
7. Test all the possible date and datetime filters for Segmentation.
